src/Pos/Wallet/Web/Error/Util.hs

```diff
@@ -36,8 +33,11 @@ catchEndpointErrors action = catchOtherError $ tryWalletError action
     tryWalletError
         | isDevelopment = try
         | otherwise     = tryJust $ \e -> (e ^? _RequestError) $> e
-    catchOtherError = handleAll $ \e -> do
-        usingLoggerName logName $
-            logError $ sformat ("Uncaught error in wallet method: "%shown) e
-        throwM err500
+    catchOtherError = flip catches
+        [ Handler $ throwM @_ @ServantErr
+        , Handler $ \(SomeException e) -> do
+            usingLoggerName logName $
+                logError $ sformat ("Uncaught error in wallet method: "%shown) e
+            throwM err500
```

Comment:
I think error body should be customized by putting formatted error into it. Here is an example: https://www.stackage.org/haddock/lts-8.13/servant-server-0.9.1.1/Servant-Server.html#v:err500